### PR TITLE
Personal goals COMPLETE overhaul

### PR DESCRIPTION
**Pull request by aidan-bowman**
## About the Pull Request

Rips out old personal goal code (which was unoptimal as well as boring) and replaces it with a new expandable and useful system.

## Why It's Good For The Game

The new system allows for more direct and integrated goals, as well as allowing for future development in antagonist systems.

## Future changes/improvements

Code could be cleaned up a bit in several areas (e.g. some signals added are kinda shitty). Antagonist objectives aren't part of the goal system yet. There also should be some player-driven goal management (e.g. rerolling goals). Admins should be given some way to manipulate goals as well.

## Changelog

:cl:
del: Deleted old (shitty) goals system.
add: Added NEW goals/objectives system! Use the "Show Goals" verb in the IC tab to view your goals. These goals are dynamically added, resolved, and rewarded mid-round.
add: Added Skills category for goals. Complete simple skill-related goals to receive a temporary buff in their respective skill.
add: Added Sanity category for goals. Complete some basic RP goals to recover sanity.
add: Weight lifting machinery added to D-class reward area.
refactor: Added some relevant signals. Old goal system hardcoded checks removed.
/:cl:

### CODE REVIEW

**Comment by aidan-bowman** [comment]
Ready for testmerge.

File: code/__defines/dcs/signals.dm [hunk 1]
```diff
@@ -23,11 +23,50 @@
...
+/// fired by a human when it despawns (e.g. by a cryopod)
+#define COMSIG_HUMAN_DESPAWNED "human_despawned"
+
+/// fired by a mob/living/carbon when it ingests a reagent. IS DYNAMICALLY CONSTRUCTED: append reagent type onto the string
+#define COMSIG_REAGENT_INGESTED "reagent_ingested"
+/// fired by something when it smokes something (as in a cigarette). should pass the object being smoked and the amount of reagents
+#define COMSIG_SMOKED_SMOKABLE "smoked_smokable"
+
+/// fired by a bank account when it has money deposited into it
+#define COMSIG_MONEY_DEPOSITED "money_deposited"
```
Comment (Tsurupeta) [inline code/__defines/dcs/signals.dm]: Add signal arguments into autodoc

File: code/game/gamemodes/game_mode.dm [hunk 2]
```diff
@@ -204,11 +204,10 @@ var/global/list/additional_antag_types = list()
...
 		antag.post_spawn()
 
-	// Update goals, now that antag status and jobs are both resolved.
+	// Post-initialize, now that antag status and jobs are both resolved.
 	for(var/thing in SSticker.minds)
-		var/datum/mind/mind = thing
-		mind.generate_goals(mind.assigned_job, is_spawning=TRUE)
-		mind.current.show_goals()
+		var/datum/mind/M = thing
+		SEND_SIGNAL(M, COMSIG_MIND_POST_INIT)
```
Comment (Tsurupeta) [inline code/game/gamemodes/game_mode.dm:210]: Bruh??? Global signals exist, wtf is this?

Comment (aidan-bowman) [reply code/game/gamemodes/game_mode.dm:210]: > Bruh??? Global signals exist, wtf is this?

that signal is also sent on latejoining mobs

File: code/__defines/dcs/signals.dm (hunk 1 above)
Comment (aidan-bowman) [reply code/__defines/dcs/signals.dm]: is there a way to do that with a 1 line comment or do i need to expand them out?

Comment (TheDarkElites) [reply code/__defines/dcs/signals.dm]: @Tsurupeta r e s p o n d

**Review by TheDarkElites — APPROVED** [review]
Tusurpta reviewed and pointed out issues have been fixed. This also passed TM, and it looks fine to me.

**Pull request merged**